### ISSUE TITLE
[Research] Lifetime of application credetials

### ISSUE THREAD

**Issue by garloff** [SovereignCloudStack/issues#316]
Do they survive the life of the user that created them?

**Comment by garloff** [comment]
@jschoone found this: https://docs.openstack.org/keystone/zed/user/application_credentials.html
“Removing a user’s role assignment on a project will invalidate the user’s application credentials for that project.”

**Comment by garloff** [comment]
So, we may need to test this and also understand if removing a local user or a federated user will actually result in a role assignment removal..

**Comment by reqa** [comment]
> @jschoone found this: https://docs.openstack.org/keystone/zed/user/application_credentials.html “Removing a user’s role assignment on a project will invalidate the user’s application credentials for that project.”

Yes, I think we need to identify some use cases/workflows, research/test and *document* the findings so that SCS devs and SCS users known how this behaves and what gotchas to look out for and address by dedicated custom tooling (if we cannot handle it on SCS/openstack level). One use case could be:

* An SCS customer (or SCS operator) federates out to an external IAM (UCS, Azure, ADFS, whatever) using an SSO protocol (OIDC, maybe SAML). An external user account has been used to access openstack APIs in SCS. Assume the user account could have been used to obtain an app credential. What does the SCS customer need to pay attention to if the user shall be disabled (or removed). Is it enough to disable the user in the customers IAM? What extra steps are required to limit continued use of possibly leaked credentials (app credentials, other CLI access tokens)? What are the default credential lifetime limits without taking any action? Can the use of those credentials be cut down or at least be monitored?
